src/fiatconnect/FiatDetailsScreen.tsx

```diff
@@ -0,0 +1,273 @@
+const SCHEMA_TO_FIELD_METADATA_MAP = {
+  AccountNumber: [
+    {
+      name: 'accountName',
+      label: i18n.t('fiatAccountSchema.accountName.label'),
+      regex: /.*?/,
+      placeholderText: i18n.t('fiatAccountSchema.accountName.placeholderText'),
+      errorMessage: i18n.t('fiatAccountSchema.accountName.errorMessage'),
+    },
+    {
+      name: 'institutionName',
+      label: i18n.t('fiatAccountSchema.institutionName.label'),
+      regex: /.*?/,
+      placeholderText: i18n.t('fiatAccountSchema.institutionName.placeholderText'),
+      errorMessage: i18n.t('fiatAccountSchema.institutionName.errorMessage'),
+    },
+    {
+      name: 'accountNumber',
+      label: i18n.t('fiatAccountSchema.accountNumber.label'),
+      regex: /^[0-9]{10}$/,
+      placeholderText: i18n.t('fiatAccountSchema.accountNumber.placeholderText'),
+      errorMessage: i18n.t('fiatAccountSchema.accountNumber.errorMessage'),
+    },
+  ],
+}
+
+const FiatDetailsScreen = ({ route, navigation }: Props) => {
+  const { t } = useTranslation()
+  const { providerURL, fiatAccountSchema } = route.params
+  const [validInputs, setValidInputs] = useState(false)
+  const [textValue, setTextValue] = useState('')
+  const [errors, setErrors] = useState(new Set<string>())
+  const inputRefs = useRef<string[]>([textValue])
+
+  const dispatch = useDispatch()
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      headerTitle: i18n.t('fiatDetailsScreen.header'),
+    })
+  }, [navigation])
+
+  const getFieldsBySchema = (fiatAccountSchema: FiatAccountSchema): Fields[] => {
+    return SCHEMA_TO_FIELD_METADATA_MAP[fiatAccountSchema]
+  }
+
+  const formFields = useMemo(() => {
+    const fields = getFieldsBySchema(fiatAccountSchema)
+
+    for (let i = 0; i < fields.length; i++) {
+      inputRefs.current.push('')
+    }
+    return fields
+  }, [fiatAccountSchema])
+
+  const onPressNext = async () => {
+    validateInput()
+
+    if (validInputs) {
+      const body: Record<string, string> = {}
+      for (let i = 0; i < formFields.length; i++) {
+        body[formFields[i].name] = inputRefs.current[i]
+      }
+
+      // TODO: some schema requires additional fields in body in addition
+      // to the ones collected from the form, eg. country code
+      await addNewFiatAccount(providerURL, fiatAccountSchema, body)
+        .then((data) => {
+          // TODO Tracking here
+          dispatch(showMessage(t('fiatDetailsScreen.addFiatAccountSuccess')))
+        })
+        .catch((error) => {
+          // TODO Tracking here
+          if (error === FiatConnectError.ResourceExists) {
+            dispatch(showError(ErrorMessages.ADD_FIAT_ACCOUNT_RESOURCE_EXIST))
+          }
+          Logger.error(TAG, `Error adding fiat account: ${error}`)
+        })
+
+      // TODO: navigate to the next screen
+    }
+  }
+
+  const onPressSelectedPaymentOption = () => {
+    // TODO: tracking here
+    // todo: naviagte to select payment options screen
```

Comment:
Why is this a TODO, you should be able to do it now right?

Reply:
coz we haven't implemented the quotes end point for FC, I didn't want to assume how amount is passed into the component